config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/EntityBuilderHelper.java

```diff
@@ -12,24 +12,50 @@
 import com.ca.apim.gateway.cagatewayconfig.util.gateway.MappingProperties;
 import com.ca.apim.gateway.cagatewayconfig.util.paths.PathUtils;
 import com.google.common.annotations.VisibleForTesting;
-import com.google.common.collect.ImmutableMap;
 import org.w3c.dom.Element;
 
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.util.HashMap;
 import java.util.Map;
+import java.util.logging.Logger;
+import java.util.regex.Pattern;
 
-import static com.ca.apim.gateway.cagatewayconfig.beans.Folder.ROOT_FOLDER_ID;
 import static com.ca.apim.gateway.cagatewayconfig.bundle.builder.Entity.*;
 import static com.ca.apim.gateway.cagatewayconfig.util.gateway.MappingActions.NEW_OR_EXISTING;
+import static com.ca.apim.gateway.cagatewayconfig.util.gateway.MappingActions.NEW_OR_UPDATE;
 import static com.ca.apim.gateway.cagatewayconfig.util.gateway.MappingProperties.*;
 
 class EntityBuilderHelper {
+    private static final Logger LOGGER = Logger.getLogger(EntityBuilderHelper.class.getName());
+    // TODO: reconsider the default value. NewOrExisting is the safest and acceptable action for most of the customer's scenarios.
```

Comment:
comment and the actual default value does not match

Reply:
It was intentional. As it is a big change, I wanted it to be tracked inside code as a TODO.